INSTALL.rst

```diff
@@ -1,51 +1,114 @@
-#########################
 Installation Instructions
-#########################
+=========================
 
-There are two options: installing locally or using within a Docker
-container.
+Let's see how to install the Circuit Knitting Toolbox (CKT). The first
+thing to do is choose how you're going to run and install the
+packages. There are three primary ways to do this: :ref:`Option 1`,
+:ref:`Option 2`, or :ref:`Option 3`.
 
-- If you are using Linux or macOS with an Intel chip (i.e., not the
-  new M1 or M2 chips), everything should work natively, so we
-  recommend the first option.
-- All users on ARM chips, as well as all Windows users, will have to
-  use the toolbox within Docker (the second option) for all features to
-  work as designed.
 
-Specifically, the following features are unavailable on the
-aforementioned platforms:
+Pre-Installation
+^^^^^^^^^^^^^^^^
+.. note::
 
-- The automatic wire cut search in the circuit cutting module
-  depends on cplex, which is only available on Intel chips and is not
-  yet available for Python 3.11.
-- The entanglement forging notebooks require pyscf, which does not
-  support Windows.
+    If a user wishes to use the circuit cutting toolbox to
+    automatically find optimized wire cuts for a circuit too large for
+    the free version of CPLEX, they should acquire a license and install
+    the `full
+    version <https://www.ibm.com/products/ilog-cplex-optimization-studio>`__.
 
-Option 1: Local installation
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+CutQC and entanglement forging users should consult the
+:ref:`Platform Support` section to determine which installation option
+is appropriate for them. Users who wish to run within a
+containerized environment may skip the pre-installation and move straight
+to :ref:`Option 3`. Otherwise, users who wish to install locally or via PyPI may 
+follow a few set of common instructions to prepare a Python environment for
+installation of CKT:
+
+First, create a minimal environment with only Python installed in it. We recommend using `Python virtual environments <https://docs.python.org/3.10/tutorial/venv.html>`__.
+
+.. code:: sh
+    
+    python3 -m venv /path/to/virtual/environment
+
+Activate your new environment.
+
+.. code:: sh
+    
+    source /path/to/virtual/environment/bin/activate
+
+Note: If you are using Windows, use the following commands in PowerShell:
+
+.. code:: sh
+    
+    python3 -m venv c:\path\to\virtual\environment
+    c:\path\to\virtual\environment\Scripts\Activate.ps1
+
+
+.. _Option 1:
+
+Option 1: Pip Installation
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Upgrade pip and install the CKT package.
+
+.. code:: sh
+
+    pip install --upgrade pip circuit-knitting-toolbox
```

Comment:
If you're going to go through the effort of upgrading pip, you should really do this _before_ using pip to install ckt.

Reply:
Yeah, that works, or you can do it on two separate lines.

A big part of the reason for upgrading pip is to get the smartest possible dependency resolution, which can only then happen once pip is run a second time.